examples/getting_started/example_config.yaml

```diff
@@ -1,146 +1,247 @@
+            rank: ${settings.cuda_env.global_rank}
+            num_replicas: ${settings.cuda_env.world_size}
+            shuffle: true
+            dataset:
+              instance_key: train_dataset
+              pass_type: BY_REFERENCE
+    collate_fn:
+      instance_key: collate_fn
+      pass_type: BY_REFERENCE
+
+val_dataset:
+  component_key: dataset
+  variant_key: packed_mem_map_dataset_continuous
+  config:
+    raw_data_path: ./data/mem_map/redpajama_v2_samples_512_test.pbin
+    block_size: ${settings.training.sequence_length}
+    sample_key:  ${settings.referencing_keys.sample_key}
 
-data:
-  sample_key: "input_ids"
-  target_key: "target_ids"
-  sequence_len: 1024
-  train_dataloader:
-    type_hint: LLMDataLoader
-    config:
-      dataloader_tag: "train"
-      # batch_size: 8 # per rank
-      num_workers: 2
-      pin_memory: true
-      shuffle: false
-      batch_sampler:
-        type_hint: BatchSampler
-        config:
-          batch_size: 8 # per rank
-          drop_last: false
-          sampler:
-            type_hint: DistributedSampler
-            config:
-              rank: ${training.global_rank}
-              num_replicas: ${training.world_size}
-              shuffle: true
-      dataset:
-        type_hint: PackedMemMapDatasetContinuous
-        config: 
-          raw_data_path: ./data/mem_map/redpajama_v2_samples_512_train.pbin
-          block_size: ${data.sequence_len}
-          sample_key: ${data.sample_key}
-      collate_fn:
-        type_hint: GPT2LLMCollator
-        config:
-          sample_key: ${data.sample_key}
-          target_key: ${data.target_key}
-  eval_dataloaders:
-    - type_hint: LLMDataLoader
+val_dataloader:
+  component_key: data_loader
+  variant_key: default
+  config:
+    num_workers: 2
+    pin_memory: true
+    shuffle: false
+    dataloader_tag: "val"
+    dataset:
+      instance_key: val_dataset
+      pass_type: BY_REFERENCE
+    batch_sampler:
+      component_key: batch_sampler
+      variant_key: default
       config:
-        dataloader_tag: "val"
-        # batch_size: 8 # per rank
-        num_workers: 2
-        pin_memory: true
-        shuffle: false
-        batch_sampler:
-          type_hint: BatchSampler
-          config: 
-            batch_size: 8 # per rank
-            drop_last: false
-            sampler:
-              type_hint: DistributedSampler
-              config:
-                rank: ${training.global_rank}
-                num_replicas: ${training.world_size}
-                shuffle: false
-        dataset:
-          type_hint: PackedMemMapDatasetContinuous
-          config: 
-            raw_data_path: ./data/mem_map/redpajama_v2_samples_512_test.pbin
-            block_size: ${data.sequence_len}
-            sample_key: ${data.sample_key}
-        collate_fn:
-          type_hint: GPT2LLMCollator
-          config: 
-            sample_key: ${data.sample_key}
-            target_key: ${data.target_key}
+        batch_size: 4
```

Comment:
should we reference the batch size from the settings?

Reply:
Yes, I changed it to `${settings.training.local_train_micro_batch_size}` in 6c86006e5c1f6efa6ea8739ffe2931919d21441a. 

Note, however, that in `config_files/training/config_lorem_ipsum.yaml`, the `batch_size` is hardcoded as well (I did not change this as it is not related to the getting started example). Also, if we decide to always use the same `batch_size` for training, validation and testing, we might want to rename `local_train_micro_batch_size` to `local_micro_batch_size`. I will leave this "for future work" :)